docs/walkthrough.md

```diff
@@ -0,0 +1,500 @@
+    - --tls-private-key-file=/var/run/serving-cert/tls.key
+    # Prometheus is running in the same pod, so you can say your Prometheus
+    # is at `localhost`
+    - --prometheus-url=http://localhost:9090
+    # relist available Prometheus metrics every 1m
+    - --metrics-relist-interval=1m
+    # calculate rates for cumulative metrics over 30s periods.  This should be *at least*
+    # as much as your collection interval.
+    - --rate-interval=30s
+    # re-discover new available resource names every 10m.  You probably
+    # won't need to have this be particularly often, but if you add
+    # additional addons to your cluster, the adapter will discover there
+    # existance at this interval
+    - --discovery-interval=10m
+    - --v=4
+    ports:
+    # this port exposes the custom metrics API
+    - containerPort: 6443
+      name: https
+      protocol: TCP
+    volumeMounts:
+    - mountPath: /var/run/serving-certs
+      name: serving-certs
+      readOnly: true
+  volumes:
+  ...
+  - name: serving-certs
+    secret:
+      secretName: serving-cm-adapter
+```
+
+</details>
+
+Next, create the deployment and expose it as a service, mapping port 443
+to your pod's port 443, on which you've exposed the custom metrics API:
+
+```shell
+$ kubectl create -f prom-adapter.deployment.yaml
+$ kubectl create service clusterip prometheus --tcp=443:6443
+```
+
+### Registering the API ###
+
+Now that you have a running deployment of Prometheus and the adapter,
+you'll need to register it as providing the
+`custom-metrics.metrics.k8s.io/v1alpha` API.
+
+For more information on how this works, see [Concepts:
+Aggregation](https://github.com/kubernetes-incubator/apiserver-builder/blob/master/docs/concepts/aggregation.md).
+
+You'll need to create an API registration record for the
+`custom-metrics.metrics.k8s.io/v1alpha1` API.  In order to do this, you'll
+need the base64 encoded version of the CA certificate used to sign the
+serving certificates you created above.  If the CA certificate is stored
+in `/tmp/ca.crt`, you can get the base64-encoded form like this:
+
+```shell
+$ base64 --w 0 < /tmp/ca.crt
+```
+
+Take the resulting value, and place it into the following file:
+
+<details>
+
+<summary>cm-registration.yaml</summary>
+
+```yaml
+apiVersion: apiregistration.k8s.io/v1beta1
+kind: APIService
+metadata:
+  name: v1alpha1.custom-metrics.metrics.k8s.io
+spec:
+  # this tells the aggregator how to verify that your API server is
+  # actually who it claims to be
+  caBundle: <base-64-value-from-above>
+  # these specify which group and version you're registering the API
+  # server for
+  group: custom-metrics.metrics.k8s.io
+  version: v1alpha1
+  # these control how the aggregator prioritizes your registration.
+  # it's not particularly relevant in this case.
+  groupPriorityMinimum: 1000
+  versionPriority: 10
+  # finally, this points the aggregator at the service for your
+  # API server that you created
+  service:
+    name: prometheus
```

Comment:
I prefer to deploy this aggregator in the `custom-metrics` ns or something like that, and have the service named `api`